src/lib.rs

```diff
@@ -0,0 +1,230 @@
+                unsafe {
+                    std::str::from_utf8_unchecked(
+                        memmap::Mmap::map(&File::open(&self.input)?)?.as_ref(),
+                    )
+                },
+                &mut BufWriter::new(File::create(&self.output)?),
+            )?;
+            uwu_progress_bar.finish_with_message("UwU'ifying Complete!");
+            Ok(())
+        }
+    }
+
+    fn uwuify_sentence<T: Write>(&mut self, text: &str, out: &mut T) -> Result<(), std::io::Error> {
+        text.lines().try_for_each(|line| {
+            line.split_whitespace()
+                .map(|f| f.as_bytes())
+                .try_for_each(|word| {
+                    let random_value = self.floating_rng.gen_range(0.0..1.0);
+
+                    if random_value <= self.faces {
+                        out.write_all(FACES[self.int_rng.gen_range(0..FACES_SIZE)])?;
+                        out.write_all(b" ")?;
+                    } else if random_value <= self.actions {
+                        out.write_all(ACTIONS[self.int_rng.gen_range(0..ACTIONS_SIZE)])?;
+                        out.write_all(b" ")?;
+                    } else if random_value <= self.stutters {
+                        (0..self.int_rng.gen_range(1..2))
+                            .into_iter()
+                            .try_for_each(|_| {
+                                match word[0] {
+                                    b'L' | b'R' => out.write_all(b"W"),
+                                    b'l' | b'r' => out.write_all(b"w"),
+                                    b => out.write_all(&[b]),
+                                }?;
+                                out.write_all(b"-")
+                            })?;
+                    }
+
+                    if self
+                        .linkify
+                        .links(unsafe { std::str::from_utf8_unchecked(word) })
+                        .count()
+                        > 0
+                        || random_value > self.words
+                    {
+                        out.write_all(word)?;
+                    } else {
+                        (0..word.len()).try_for_each(|index| match word[index] {
+                            b'L' | b'R' => out.write_all(b"W"),
+                            b'l' | b'r' => out.write_all(b"w"),
+                            b'E' | b'e' | b'A' | b'I' | b'O' | b'U' | b'a' | b'i' | b'o' | b'u' => {
+                                match word.get(index - 1).unwrap_or(&word[0]) {
+                                    b'N' | b'n' => out.write_all(&[b'y', word[index]]),
+                                    _ => out.write_all(&[word[index]]),
+                                }
+                            }
+                            b => out.write_all(&[b]),
+                        })?;
+                    }
+                    out.write_all(b" ")
+                })?;
+            out.write_all(b"\n")
+        })
+    }
+}
+
+#[cfg(test)]
```

Comment:
Probably worth adding a test including non utf-8 compliant strings

Reply:
The label is pretty deceptive. Here I wrote a **benchmark**, and not a **test**. Benchmarks test the performance of your program while tests test the soundness of your program.